### PR TITLE
* ZF3 compatibility added, dropping support for ZF2

### PR DESCRIPTION
**Pull request by adrian-ziolkowski**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
 language: php
 
 php:
-  - 5.3
-  - 5.4
-  - 5.5
```
Comment (snapshotpl) [inline .travis.yml]: Missing test with composer with lower dependencies

File: README.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ the root of your project:
 ```
 {
     "require": {
-        "slm/locale": ">=0.1.0,<1.2.0-dev"
+        "slm/locale": "^1.0"
```
Comment (snapshotpl) [inline README.md]: after this changes we need to break compatibility, so major version will be changed.

File: src/SlmLocale/Service/LocaleMenuViewHelperFactory.php [hunk 3]
```diff
@@ -40,23 +40,22 @@
...
      * @return LocaleMenu
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container)
     {
-        $detector = $serviceLocator->getServiceLocator()->get('SlmLocale\Locale\Detector');
+        $detector = $container->get(SlmLocale\Locale\Detector::class);
 
         $helper = new LocaleMenu;
         $helper->setDetector($detector);
```
Comment (snapshotpl) [inline src/SlmLocale/Service/LocaleMenuViewHelperFactory.php]: Looks like required dependency, but we can omit build this object without it. Maybe should be injected into constructor?

File: src/SlmLocale/Strategy/StrategyPluginManager.php [hunk 4]
```diff
@@ -40,41 +40,51 @@
...
      * {@inheritDocs}
      */
-    protected $invokableClasses = array(
-        'cookie'         => 'SlmLocale\Strategy\CookieStrategy',
-        'host'           => 'SlmLocale\Strategy\HostStrategy',
-        'acceptlanguage' => 'SlmLocale\Strategy\HttpAcceptLanguageStrategy',
-        'query'          => 'SlmLocale\Strategy\QueryStrategy',
-        'uripath'        => 'SlmLocale\Strategy\UriPathStrategy',
+    protected $aliases = array(
+        'Cookie'         => CookieStrategy::class,
```
Comment (snapshotpl) [inline src/SlmLocale/Strategy/StrategyPluginManager.php:56]: create aliases passing old names for maximum backward compatibility

File: src/SlmLocale/Strategy/StrategyPluginManager.php [hunk 5]
```diff
@@ -40,41 +40,51 @@
...
+            ));
         }
+    }
 
-        throw new Exception\InvalidStrategyException(sprintf(
-            'Plugin of type %s is invalid; must implement %s\StrategyInterface',
-            (is_object($plugin) ? get_class($plugin) : gettype($plugin)),
-            __NAMESPACE__
-        ));
+    public function validatePlugin($instance)
```
Comment (snapshotpl) [inline src/SlmLocale/Strategy/StrategyPluginManager.php]: what is difference with `validatePlugin` and `validate` method?

**Comment by snapshotpl** [comment]
@adrian-ziolkowski ping

File: src/SlmLocale/Strategy/StrategyPluginManager.php (hunk 5 above)
Comment (adrian-ziolkowski) [reply src/SlmLocale/Strategy/StrategyPluginManager.php]: validatePlugin was keept for backwords compatibility with ZF2, actually we don't need it any more (we breake BC in this release), removed

File: src/SlmLocale/Strategy/StrategyPluginManager.php (hunk 4 above)
Comment (adrian-ziolkowski) [reply src/SlmLocale/Strategy/StrategyPluginManager.php:56]: done

**Comment by adrian-ziolkowski** [comment]
@snapshotpl can we marge it?

**Comment by snapshotpl** [comment]
@adrian-ziolkowski yes, please :)

**Pull request merged**